src/flatcollections-array/FlatArray/FlatArray.T/FlatArray.Functional.Map.cs

```diff
@@ -0,0 +1,42 @@
+namespace System;
+
+partial struct FlatArray<T>
+{
+    public FlatArray<TResult> Map<TResult>(Func<T, TResult> map)
+    {
+        _ = map ?? throw new ArgumentNullException(nameof(map));
+
+        if (length == default)
+        {
+            return default;
+        }
+
+        var resultItems = new TResult[length];
+
+        for (int i = 0; i < length; i++)
+        {
+            resultItems[i] = map.Invoke(items![i]);
+        }
+
+        return new(resultItems, default);
+    }
+
+    public FlatArray<TResult> Map<TResult>(Func<T, int, TResult> map)
+    {
+        _ = map ?? throw new ArgumentNullException(nameof(map));
+
+        if (length == default)
+        {
+            return default;
+        }
+
+        var resultItems = new TResult[length];
+
+        for (int i = 0; i < length; i++)
+        {
+            resultItems[i] = map.Invoke(items![i], i);
+        }
```

Comment:
Consider merging actual `main` into this one to make it replaceable by:

```suggestion
        InternalForEach((i, item) => resultItems[i] = map.Invoke(i, item));
```

Reply:
Fixed